lib/ripper-tags/emacs_formatter.rb

```diff
@@ -74,5 +93,68 @@ def format(tag, name_field = :name)
         0,
       ]
     end
+
+    def save_rest(&block)
+      @original.save_rest(&block) if @original
+    end
+
+    def save_section(filename, &block)
+      @original.save_section(filename, &block) if @original
+    end
+
+    def remove_saved(tag)
+      @original.remove(tag) if @original
+    end
+  end
+
+  class EmacsTagsProcessor
+    def initialize(source)
+      @source = source
+      @tags = {}
+    end
+    
+    TAG_PATTERN = /^(.*)\x7F(.*)\x01(\d+),\d+/
+
+    def read
+      lines = File.readlines(@source, "\n")
+      filename = nil
+      while !lines.empty?
+        line = lines.shift
```

Comment:
Could we avoid mutating the `lines` array while iterating over it?

Reply:
Not sure why, this is just using lines array as a queue, shifting items off the front.